lib/services/nfc.dart

```diff
@@ -91,7 +92,18 @@ class NFCService {
   }
 
   NFCService() {
-    Timer(Duration(seconds: 2), _listenLnLinks);
+    NfcPlugin nfcPlugin = NfcPlugin();
+    int fnCalls = 0;
+    _checkNfcStartedWithTimer =
+        Timer.periodic(Duration(milliseconds: 100), (Timer t) {
+      if (fnCalls == 5) {
+        _checkNfcStartedWithTimer.cancel();
```

Comment:
We should also cancel the timer when we get a valid payload otherwise risking in duplicate events.

Reply:
[Done](https://github.com/breez/breezmobile/pull/341/commits/32f45fe70cdca469dddb48e164a0d41162b05271)